src/plugins/xerces/deserializer.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file
+ *
+ * @brief deserialization implementation for xerces plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include "deserializer.hpp"
+#include "util.hpp"
+
+#include <xercesc/dom/DOM.hpp>
+#include <xercesc/dom/DOMAttr.hpp>
+#include <xercesc/dom/DOMDocument.hpp>
+#include <xercesc/dom/DOMImplementation.hpp>
+#include <xercesc/dom/DOMNode.hpp>
+#include <xercesc/parsers/XercesDOMParser.hpp>
+
+#include <kdblogger.h>
+#include <key.hpp>
+
+XERCES_CPP_NAMESPACE_USE
+using namespace std;
+using namespace kdb;
+using namespace xerces;
+
+namespace
+{
+
+XercesPtr<DOMDocument> doc2dom (std::string const & src)
+{
+	XercesDOMParser parser;
+	parser.setValidationScheme (XercesDOMParser::Val_Auto);
+	parser.setDoNamespaces (false);
+	parser.setDoSchema (false);
+	parser.setCreateEntityReferenceNodes (false);
+
+	parser.parse (asXMLCh (src));
+	return XercesPtr<DOMDocument> (parser.adoptDocument ());
+}
+
+string getElementText (DOMNode const * parent)
+{
+	string str;
+
+	for (auto child = parent->getFirstChild (); child != NULL; child = child->getNextSibling ())
+	{
+		if (DOMNode::TEXT_NODE == child->getNodeType () || DOMNode::CDATA_SECTION_NODE == child->getNodeType ())
+		{
+			DOMText * data = dynamic_cast<DOMText *> (child);
+			if (!data->getIsElementContentWhitespace ()) str += toStr (data->getData ());
+		}
+	}
+	// trim whitespace
```

Comment:
you never fail to point out erroneous parts :)

Now that i've looked closer at it, i found that this is quite a problem where i have no general idea of how to fix it. 
The issue is that if there is no schema or DTD provided, xerces does not now wheter certain whitespace can be safely ignored (e.g. produced by pretty printing) or if it is indeed part of the value. Consider the following XML fragment, which is obviously pretty printed
```
<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<xerces>  value of<!-- Comment inside text --> xerces  
	<!-- this is an attribute -->
	<fizz buzz="fizzBuzz" />
	<userKey user="key">withValue</userKey>
	<!-- those need escaping: " ' < > &  -->
	<späciöl_-keüs1 attr='"' attr2="$%(){}``äüö²[/\'>&quot;&lt;&apos;&amp;">>"&amp;&lt;'</späciöl_-keüs1>
	<cdata more-s_päcials="1 &amp; 2 are &lt; &#51; &#x0A;"><![CDATA[this is some cdata text "'<>&ä "]]></cdata>
</xerces>
```

Now the line of the xerces element contains the text `  value of xerces  ` (note the two intended spaces). But actually there are much more text nodes under the xerces element than only the first one, most of them containing only the tabs and newlines for pretty printing. So when appending all the text nodes together, we have a lot of newlines and tabs caused by pretty printing which we don't want to have in our output in that example.

So i though about just picking the first text node, but this would fail if we put the text after the other child elements, or have multiple text nodes between other element nodes (which is allowed in xml).

My other idea was the trimming which i did before, but you are right it may lead to unwanted behavior too. I could ignore text nodes which only contain whitespace as those are most likely caused by pretty printing, but this may result in some false positives. And the way you can pretty print an xml file may vary, so its really hard to make assumptions here.

If there is some XML schema or DTD provided, xerces can deduce from the schema if a given text node is just unnecessary whitespace or actually a relevant node and this should prevent such situations. But as a lot of XML documents probably have no schema or DTD, we need a good default behavior for this. Pretty printed documents are fairly common, but the way they are pretty printed may vary.